src/adapter/src/coord/sequencer/inner/peek.rs

```diff
@@ -962,6 +1042,52 @@ impl Coordinator {
         Ok(Self::send_immediate_rows(rows))
     }
 
+    #[tracing::instrument(level = "debug", skip_all)]
+    async fn peek_stage_copy_to_finish(
+        &mut self,
+        ctx: &mut ExecuteContext,
+        PeekStageCopyToFinish {
+            validity: _,
+            optimizer,
+            global_lir_plan,
+            copy_to_context,
+        }: PeekStageCopyToFinish,
+    ) -> Result<ExecuteResponse, AdapterError> {
+        let session = ctx.session_mut();
+        let (peek_plan, df_meta) = global_lir_plan.unapply();
+        let PeekPlan::SlowPath(PeekDataflowPlan {
+            desc: dataflow,
+            id: _,
+            ..
+        }) = peek_plan
+        else {
+            unreachable!()
+        };
+
+        self.emit_optimizer_notices(&*session, &df_meta.optimizer_notices);
+
+        // Ship the dataflow
+        let output_ids = dataflow.export_ids().collect();
+        // Very important: actually create the dataflow (here, so we can destructure).
+        self.controller
+            .active_compute()
+            .create_dataflow(optimizer.cluster_id(), dataflow)
+            .unwrap_or_terminate("cannot fail to create dataflows");
+        self.initialize_compute_read_policies(
+            output_ids,
+            optimizer.cluster_id(),
+            // Disable compaction so that nothing can compact before the peek occurs below.
+            CompactionWindow::DisableCompaction,
+        )
+        .await;
```

Comment:
I have this from the original peek implementation. I am not sure if I need this in COPY TO since there's not going to be a peek call after. Does that sound right?

cc @petrosagg / @aalexandrov

Reply:
I think this depends on the dataflow internals. Since the actual export occurs in a dataflow operator, maybe...you don't need this because the dataflow is able to hold back whatever it needs?

Reply:
That's what I thought too (I am fuzzy on how dataflow holds back stuff). Removing this for the time being.

Reply:
IIRC, this happens with the various `.downgrade` calls on capabilities. Timestamps are held back until explicitly released.

Reply:
For a sink it doesn't make sense to try to install a read hold on its output. A sink output cannot be read (at least not in the scope of compute), only an index can. In fact, https://github.com/MaterializeInc/materialize/pull/24487, which I want to merge tomorrow morning will make it an error to try to set a read policy on a non-index compute collection.